src/parse.rs

```diff
@@ -105,21 +88,49 @@ mod tests {
     use super::*;
 
     #[test]
-    fn test_get_file_extension_from_diff_line() {
+    fn test_get_file_extension_from_marker_line() {
         assert_eq!(
-            get_file_extension_from_diff_line("diff --git a/src/main.rs b/src/main.rs"),
+            get_file_extension_from_marker_line(
+                "--- src/one.rs	2019-11-20 06:47:56.000000000 +0100"
+            ),
```

Comment:
So I believe that there's another bug remaining (not your fault or responsibility!), in that unified diff is still broken for file names with spaces:

```diff
--- "src/pa r se.rs"	2020-05-03 16:40:13.851935846 -0400
+++ "src/pa rse.rs"	2020-05-03 16:41:49.798457052 -0400
@@ -231,4 +231,7 @@
         );
     }
 }
-vjhvjj
+
+pub fn test(s: &str) {
+    println!("hello {}", s);
+}
```

<table><tr><td><img width=400px src="https://user-images.githubusercontent.com/52205/80925367-3b06d580-8d5d-11ea-9d3d-7c569456ca0d.png" alt="image" /></td></tr></table>

If I remove the double quotes and spaces from those filenames, then delta syntax-highlights the added function as Rust:

```diff
--- src/parse.rs	2020-05-03 16:40:13.851935846 -0400
+++ src/parse.rs	2020-05-03 16:41:49.798457052 -0400
@@ -231,4 +231,7 @@
         );
     }
 }
-vjhvjj
+
+pub fn test(s: &str) {
+    println!("hello {}", s);
+}
```

<table><tr><td><img width=400px src="https://user-images.githubusercontent.com/52205/80925410-87521580-8d5d-11ea-945c-65843c123dc5.png" alt="image" /></td></tr></table>